### ISSUE TITLE
Fix slowdown in new version

### ISSUE THREAD

**Issue by bootandy** [bootandy/dust#415]
Tests with hyperfine show that this PR means we are running ~ 20% slower than the v1.0.0 release. 

https://github.com/bootandy/dust/pull/399

**Comment by bootandy** [comment]
Ok, I've traced the problem down to 2 places. 

If no filters are entered we always apply this filter:
`   None => (Operater::GreaterThan, 0),`

Moving this line down in dir_walker.rs (it is needed because having it is faster than a file lookup).
`if !walk_data.allowed_filesystems.is_empty()`

**Comment by bootandy** [comment]
https://github.com/bootandy/dust/pull/417

**Comment by wugeer** [comment]
> Ok, I've traced the problem down to 2 places.
>
> If no filters are entered we always apply this filter: - meaning we always have to check it which we never did before , which we've never done before `None => (Operater::GreaterThan, 0),`
>
> The Moving of this line down a bit in dir_walker.rs (it is needed because having it is faster than a file lookup). It's faster than file lookup). `if !walk_data.allowed_filesystems.is_empty()`

Yes, you are right.  I'm sorry that this PR of mine caused the performance to drop so much. I will revisit the performance of my code!

**Comment by bootandy** [comment]
No problem, We live and learn. I should have checked more.
